activerecord/lib/active_record/validations/presence.rb

```diff
@@ -5,7 +5,9 @@ module Validations
     class PresenceValidator < ActiveModel::Validations::PresenceValidator # :nodoc:
       def validate_each(record, attribute, association_or_value)
         if record.class._reflect_on_association(attribute)
-          association_or_value = Array.wrap(association_or_value).reject(&:marked_for_destruction?)
+          association_or_value = Array.wrap(association_or_value)
```

Comment:
I don't think this is the correct fix. We are duplicating the super class behavior here. The correct fix would be to change the null object to define to define a `to_ary` that returns an empty array.